snapcraft/application.py

```diff
@@ -213,6 +234,7 @@ def _setup_partitions(self, yaml_data: dict[str, Any]) -> list[str] | None:
         # This is why we have the enablement after the check, if not
         # we could have directly returned with .get_partitions() which
         # handles the empty case.
+        craft_parts.Features.reset()
```

Comment:
This is another upstream nuisance.  Before this runs, craft-application checks for features and unintentionally defines an empty feature set:

https://github.com/canonical/craft-application/blob/1509cec9af705c762b06725bf867ef1d9e1d822b/craft_application/commands/lifecycle.py#L44-L45

Reply:
I think in this case you can/should use [_enable_craft_parts_features()](https://github.com/canonical/craft-application/blob/main/craft_application/application.py#L760)

Reply:
Ideally we should never call feature reset in production code (it logs a warning, and was primarily designed for testing convenience).

Reply:
Thanks, that is a clean fix!